ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/CommandParser.java

```diff
@@ -0,0 +1,228 @@
+import io.confluent.ksql.execution.expression.tree.CreateMapExpression;
+import io.confluent.ksql.execution.expression.tree.CreateStructExpression;
+import io.confluent.ksql.execution.expression.tree.DecimalLiteral;
+import io.confluent.ksql.execution.expression.tree.DoubleLiteral;
+import io.confluent.ksql.execution.expression.tree.Expression;
+import io.confluent.ksql.execution.expression.tree.IntegerLiteral;
+import io.confluent.ksql.execution.expression.tree.LongLiteral;
+import io.confluent.ksql.execution.expression.tree.StringLiteral;
+import io.confluent.ksql.metastore.TypeRegistry;
+import io.confluent.ksql.parser.AstBuilder;
+import io.confluent.ksql.parser.DefaultKsqlParser;
+import io.confluent.ksql.parser.KsqlParser;
+import io.confluent.ksql.parser.tree.InsertValues;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public final class CommandParser {
+  private static final String QUOTED_STRING_OR_WHITESPACE = "('([^']*|(''))*')|\\s+";
+  private static final KsqlParser KSQL_PARSER = new DefaultKsqlParser();
+  private static final String INSERT = "INSERT";
+  private static final String INTO = "INTO";
+  private static final String VALUES = "VALUES";
+  private static final String SELECT = "SELECT";
+  private static final String CREATE = "CREATE";
+  private static final String SINK = "SINK";
+  private static final String SOURCE = "SOURCE";
+  private static final String CONNECTOR = "CONNECTOR";
+  private static final String SHORT_COMMENT_OPENER = "--";
+  private static final String SHORT_COMMENT_CLOSER = "\n";
+  private static final String LONG_COMMENT_OPENER = "/*";
+  private static final String LONG_COMMENT_CLOSER = "*/";
+  private static final char SINGLE_QUOTE = '\'';
+  private static final char SEMICOLON = ';';
+
+  private CommandParser() {
+  }
+
+  public static List<SqlCommand> parse(final String sql) {
+    return splitSql(sql).stream()
+        .map(CommandParser::transformToSqlCommand)
+        .collect(Collectors.toList());
+  }
+
+  /*
+  * Splits a string of sql commands into a list of commands and filters out the comments.
+  *
+  * @return a list of strings
+  * */
+  @VisibleForTesting
+  static List<String> splitSql(final String sql) {
+    final List<String> commands = new ArrayList<>();
+    String current = "";
+    for (int i = 0; i < sql.length(); i++) {
+      if (sql.charAt(i) == SINGLE_QUOTE) {
+        final int closingToken = sql.indexOf(SINGLE_QUOTE, i + 1);
+        current += sql.substring(i, closingToken + 1);
+        i = closingToken;
+      } else if (i < sql.length() - 1 && sql.startsWith(SHORT_COMMENT_OPENER, i)) {
+        i = sql.indexOf(SHORT_COMMENT_CLOSER, i + 1);
+      } else if (i < sql.length() - 1 && sql.startsWith(LONG_COMMENT_OPENER, i)) {
+        i = sql.indexOf(LONG_COMMENT_CLOSER, i + 1) + 1;
```

Comment:
Manipulating `i` inside the for-loop feels error-prone, and also requires reasoning about offsets like this. (This hard-coded `+ 1` should more accurately be `+ LONG_COMMENT_CLOSER.length - 1`, and similarly for the others.) Simplify the code by using a while loop instead?